File: catkin_ws/src/my_robot_tutorials/scripts/add_two_ints_client.py
#!/usr/bin/env python3

# This Python script implements a ROS service client that calls the 'add_two_ints' service.
# It uses the AddTwoInts service provided by the rospy_tutorials package.

import rospy
from rospy_tutorials.srv import AddTwoInts

if __name__ == '__main__':
    # Initialize a ROS node and name it
    rospy.init_node('add_two_ints_client')

    # Wait for the 'add_two_ints' service to become available
    rospy.wait_for_service("/add_two_ints")


  
    try:
        add_two_ints = rospy.ServiceProxy("/add_two_ints", AddTwoInts)   # Create a service proxy for the 'add_two_ints' service
        response = add_two_ints(2, 6)                                    # Call the service and get the response
        rospy.loginfo("Sum is: " + str(response.sum))                    # Log the sum of the two integers
      
    except rospy.ServiceException as e:
        rospy.logwarn("Service FAILED: " + str(e))                        # Log a warning if the service call fails

File: catkin_ws/src/my_robot_tutorials/src/add_two_ints_client.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>



int main(int argc, char **argv) {
  
  ros::init(argc, argv, "add_two_ints_client");   // Initialize the ROS node with the name 'add_two_ints_client'
  ros::NodeHandle nh;                             // Create a node handle to interact with ROS

  // Create a service client to call the 'add_two_ints' service
  ros::ServiceClient client = nh.serviceClient<rospy_tutorials::AddTwoInts>("/add_two_ints");

  // Prepare the request for the 'add_two_ints' service
  rospy_tutorials::AddTwoInts srv;
  srv.request.a = 12;
  srv.request.b = 5;



  
  // Call the 'add_two_ints' service and process the response
  if (client.call(srv)) {
    // The service call was successful, so process the response
    ROS_INFO("Returned Sum is : %d", (int)srv.response.sum);

  } else {
    // The service call failed, so print a warning
    ROS_WARN("Service Call FAILED");
  }

  
}
